src/lib.rs

```diff
@@ -297,7 +316,7 @@ type Result<T> = core::result::Result<T, Error>;
 type StreamResult<T> = core::result::Result<T, StreamError>;
 
 
-#[derive(Clone, Copy, PartialEq)]
+#[derive(Clone, Copy, PartialEq, Debug)]
```

Comment:
Not necessary. Private API.

Reply:
This is needed to derive Debug on Tokenizer

Reply:
You don't need `Debug` on `Tokenizer` either.

Reply:
It is a public type, and hence should implement `Debug` pretty liberally. (I needed it in my use of your library)

Reply:
Why would you want to debug an iterator?

Reply:
I oppose it because it has no value.

I would suggest simply adding 

```rust
impl core::fmt::Debug for Tokenizer<'_> {
    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
        write!(f, "Tokenizer {{ ... }}")
    }
}
```